tripy/tripy/frontend/trace/ops/shape.py

```diff
@@ -63,4 +63,8 @@ def shape(self: "tripy.Tensor") -> List["tripy.DimensionSize"]:
         assert shape == [8, 2]
     """
 
+    # If the shape is statically known, we do not need to insert any operator calls.
+    if all(dim >= 0 for dim in self.trace_tensor.shape):
```

Comment:
Oh one more thing here is that we don't want to do this if we're tracing for compilation. Can you add another condition: `and not self.trace_tensor.is_compile_tracer`?

Reply:
The reason is that we may want to use dynamic shapes at runtime that are different than the shape of the tensor that's being passed in for tracing. In that case we have to insert operations to get the shape at runtime.

Reply:
Why is `compile_tracer` being set for the parameter update? That should only happen in the `compile` call.

Reply:
My mistake, I was assuming that it was tracing based on having a difference between profiling results but it seems to have just been variability on my system. There was no difference when I checked again and I put in an assert to verify that the compile tracer wasn't being used for that case.

Reply:
Could we also add a test case for this? We could have some code like:
```py
compiled_ones = tp.compile(tp.ones_like, args=[...]) # Use an input with a dynamic shape here

a = tp.Tensor(...)
out = compiled_ones(a.shape) 
assert out.shape == a.shape # Test with various shapes of `a` and make sure it works
```
and then compile it with dynamic shapes. If we accidentally freeze the shapes while compiling, this should catch it.

Reply:
This particular test case works without the additional condition as well. Ones I try where the shape is used in the compiled function still all pass without the extra condition, since the dims for the trace tensor are all -1 regardless of the input info.

Per compile.py, it doesn't seem that the shape bounds specified in the input info will make it to the individual ops:
```
    def process_arg(name, arg):
        if isinstance(arg, InputInfo):
            # Make new tensors for tracing.
            from tripy.common.datatype import floating, integer
            from tripy.frontend.trace.ops.fill import full

            init_value = 1 if issubclass(arg.dtype, integer) else 1.0 if issubclass(arg.dtype, floating) else True
            tensor = full(shape=arg.shape_bounds.opt, value=init_value, dtype=arg.dtype) # arg.shape_bounds.opt comes from input info
            tensor.name = name
            tensor.trace_tensor.is_compile_tracer = True
            # however, tensor.trace_tensor.shape is all -1

            trace_input_map[name] = tensor
            shapes.append(arg.shape_bounds)
            input_names.add(name)

            return tensor
        return arg
```

Reply:
Ah that must be because we use `full` to generate the input tensors for tracing. I think it's still a good test to have to ensure we don't make changes to how we generate input tensors that will break dynamic shapes.